### PR TITLE
[2.0] Add _encryption_algorithm_to_crypt_instance method.

### PR DESCRIPTION
**Pull request by bantu**

### CODE REVIEW

**Comment by terrafrost** [comment]
Looks good to me! Altho it does seem like this could target 1.0 instead of 2.0? #754 targeted 1.0 for instance.

**Comment by bantu** [comment]
It was intentional due to object handling (copying) in PHP4.

**Comment by terrafrost** [comment]
Ah okay. Well, looks good to me!

File: phpseclib/Net/SSH2.php [hunk 1]
```diff
@@ -1894,6 +1802,50 @@ function _encryption_algorithm_to_key_size($algorithm)
...
+            case 'twofish128-ctr':
+            case 'twofish192-ctr':
+            case 'twofish256-ctr':
+                return new Twofish(Base::MODE_CTR);
+            case 'arcfour':
+            case 'arcfour128':
+            case 'arcfour256':
+                return new RC4();
+        }
+        return null;
```
Comment (GrahamCampbell) [inline phpseclib/Net/SSH2.php]: This shouldn't be needed, if a method has no return statement, PHP treats it as `null`.

Comment (bantu) [inline phpseclib/Net/SSH2.php]: True. Is there a sniff for that?

Comment (GrahamCampbell) [inline phpseclib/Net/SSH2.php]: Not sure, I don't use phpcs, sorry.

**Pull request merged**